tests/unit/llms/test_llm_loader.py

```diff
@@ -131,28 +130,16 @@ def mock_import(module, *args, **kwargs):
             LLMLoader(provider=provider, model=constants.GRANITE_13B_CHAT_V1)
 
 
-def _prepare_openapi_config():
-    provider_config = ProviderConfig()
-    provider_config.models = {constants.GRANITE_13B_CHAT_V1: "mock model"}
-    config.llm_config.providers = {constants.PROVIDER_OPENAI: provider_config}
-
-
-@patch.dict(os.environ, {"OPENAI_API_KEY": ""})
 def test_constructor_openai_llm_instance_no_api_key():
     """Test the construction fo LLM instance for OpenAI when API key is not provided."""
-    _prepare_openapi_config()
-
     # no API key is provided so validation should fail
+    config.init_config("tests/config/without_openai_api_key.yaml")
     with pytest.raises(Exception, match="Did not find openai_api_key"):
```

Comment:
this error message doesn't seem specific enough.  It seems like this test is checking what happens when the api key file is provided, but empty.

There are really 3 scenarios to test:
1) no filepath is provided
2) filepath is provided, but the file does not exist
3) filepath is provided, but the file is empty

we should ensure each of those scenarios has a unique/obvious error message specific to the situation.

Reply:
This tests the LLM loader, the "Did not find openai_api_key" message comes from langchain, https://github.com/langchain-ai/langchain/blob/cc173344737d4ac7ff96a04bfe2e93022b32d8b7/libs/core/langchain_core/utils/env.py#L42 most likely.
1. Credentials are optional in config: https://github.com/openshift/lightspeed-service/blob/00a3357045fd6aa7c3aaa242b30ccdfccf6cde60/ols/app/models/config.py#L48. There're many tests in https://github.com/openshift/lightspeed-service/blob/main/tests/unit/utils/test_config.py that don't specify credentials.
2. We have tests like that: https://github.com/openshift/lightspeed-service/blob/00a3357045fd6aa7c3aaa242b30ccdfccf6cde60/tests/unit/utils/test_config.py#L267 and https://github.com/openshift/lightspeed-service/blob/00a3357045fd6aa7c3aaa242b30ccdfccf6cde60/tests/unit/utils/test_config.py#L279
3. This case is not handled in the config code and but this test ^^^ does exactly that and langchain complains.

Reply:
ok, thanks